client/ayon_maya/plugins/publish/validate_skeleton_top_group_hierarchy.py

```diff
@@ -25,18 +37,38 @@ def process(self, instance):
         if not self.is_active(instance.data):
             return
 
-        invalid = []
-        skeleton_mesh_data = instance.data("skeleton_mesh", [])
-        if skeleton_mesh_data:
-            invalid = self.get_top_hierarchy(skeleton_mesh_data)
-            if invalid:
-                raise PublishValidationError(
-                    "The skeletonMesh_SET includes the object which "
-                    "is not at the top hierarchy: {}".format(invalid))
-
-    def get_top_hierarchy(self, targets):
-        targets = cmds.ls(targets, long=True)  # ensure long names
-        non_top_hierarchy_list = [
-            target for target in targets if target.count("|") > 2
-        ]
-        return non_top_hierarchy_list
+        skeleton_mesh_nodes = instance.data("skeleton_mesh", [])
+        if not skeleton_mesh_nodes:
+            return
+
+        invalid = get_non_root_nodes(skeleton_mesh_nodes)
+        if invalid:
+            raise PublishValidationError(
+                "The skeletonMesh_SET includes the object which "
+                "is not at the top hierarchy: {}".format(invalid))
+
+
+class ValidateAnimatedRigTopGroupHierarchy(plugin.MayaInstancePlugin,
+                                           OptionalPyblishPluginMixin):
+    """Validates top group hierarchy in the SETs
+    Make sure the object inside the SETs are always top
+    group of the hierarchy
+    """
+    order = ValidateContentsOrder + 0.05
+    label = "Animated Rig Top Group Hierarchy"
+    families = ["animation.fbx"]
+    optional = True
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+
+        skeleton_anim_nodes = instance.data("animated_skeleton", [])
+        if not skeleton_anim_nodes:
+            return
```

Comment:
@moonyuet So I've restructured the code to show the issue (and to clean up).

Anyway, here it allows the instance to be valid even when no animated skeleton is present.

Which allows this to be valid:
![image](https://github.com/user-attachments/assets/3f0f8f4f-f215-4c67-8d36-ed9cdc2037ca)

Which when **FBX** is the only product to be exported for the animation instance:
![image](https://github.com/user-attachments/assets/49e69c3f-7146-457c-99fb-e0d29f5e3825)

It will not get invalidated but it **will** error on integrate because no Extraction occurred for the FBX.
The FBX extractor reports:

```
DEBUG: Top group of animated skeleton not found in ['char_hero_rigMain_01_:rigMain_skeletonAnim_SET'].
Skipping fbx animation extraction.
```

Hence it errors on integrate that there is nothing to integrate.

![image](https://github.com/user-attachments/assets/19beede7-fb6a-415c-ae9b-1b8deceb5fb2)

[This](https://github.com/ynput/ayon-maya/blob/06aa2445b4df511225b6795b283dd974040c0b5f/client/ayon_maya/plugins/publish/extract_fbx_animation.py#L39-L50) is where it gets silently skipped... 

Which raises the question - that _if_ FBX is enabled... should this skeleton existence be enforced? A tricky question, because I checked in our productions it seems that _Collect FBX Animation_ is enabled (even though we never wanted any FBX to begin with) but that would mean that due to that being enabled... that suddenly all our instances would error because of course they are not configured to export FBX.

The question now is, what is the best way to fix this?

- How can we best enforce that when FBX is enabled, that it actually writes out something.
- And, if apparently production scenarios currently in-use apparently defaulted to having FBX enabled but silently ignored... how can we make those production scenarios safely transition?

Reply:
Is there a reason the FBX is enabled by default yet allows to be 'silently' skipped currently?

Reply:
> Is there a reason the FBX is enabled by default yet allows to be 'silently' skipped currently?

Guess we shouldn't do this anymore (We used to do this as it is optional for users, but now it is different case)

Reply:
**But** - raising an error now, suddenly, would affect ALL running productions in which it was silently ignored. For example, we suddenly couldn't publish from all animation scenes running in our AYON productions unless we start disabling the Collect FBX Animation which would be very unfortunate (and a toll on production).

Anyway, with the logic removed I now face this error in the Extractor:
```python
Traceback (most recent call last):
  File "C:\Users\User\AppData\Local\Ynput\AYON\dependency_packages\ayon_2406251801_windows.zip\dependencies\pyblish\plugin.py", line 528, in __explicit_process
    runner(*args)
  File "E:\dev\ayon-maya\client\ayon_maya\plugins\publish\extract_fbx_animation.py", line 40, in process
    namespace = get_namespace(out_members[0])
IndexError: list index out of range
```
Which is better than the integrator - but it'd still break our running productions.

Reply:
**Nevermind** - Collect FBX animation is OFF by default I just happened to look by accident at an animation publish an artist was making... that was still running OpenPype for an old project. Sorry!